From acee9409a1751a76cb95ef380be6d57c75690a45 Mon Sep 17 00:00:00 2001
From: Mickey Sola <msola@sourcefire.com>
Date: Mon, 30 Oct 2017 16:39:54 -0400
Subject: bb11941 - fixing UAF in mbox exportBounceMessage. Original patch
 submitted by Suleman Ali

Patch-Name: bb11941-fixing-UAF-in-mbox-exportBounceMessage.-Orig.patch
---
 libclamav/mbox.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libclamav/mbox.c b/libclamav/mbox.c
index 13edb78..3df2ae0 100644
--- a/libclamav/mbox.c
+++ b/libclamav/mbox.c
@@ -2053,8 +2053,9 @@ parseEmailBody(message *messageIn, text *textIn, mbox_ctx *mctx, unsigned int re
 		/*
 		 * Look for uu-encoded main file
 		 */
-		if((encodingLine(mainMessage) != NULL) &&
-		   ((t_line = bounceBegin(mainMessage)) != NULL))
+		if(mainMessage->body_first != NULL &&
+			(encodingLine(mainMessage) != NULL) &&
+			((t_line = bounceBegin(mainMessage)) != NULL))
 			rc = (exportBounceMessage(mctx, t_line) == CL_VIRUS) ? VIRUS : OK;
 		else {
 			bool saveIt;
